packages/cadl-ranch-specs/http/special-headers/request-id/main.tsp

```diff
@@ -0,0 +1,59 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Different request id header configurations.")
+@supportedBy("dpg")
+@scenarioService("/special-headers/request-id")
+namespace SpecialHeaders.RequestId;
+
+@client
+namespace DefaultClient {
+  @scenario
+  @scenarioDoc("""
+Default situation: if no request id header configured, we use standard request id header.
+""")
+  @doc("""
+Default request id header.
+""")
+  @post
+  @route("/default")
+  op default(): NoContentResponse;
+}
+
+@client
+namespace StandardClient {
+  @scenario
+  @scenarioDoc("""
+Standard request id header. SDK should not genreate `requestId` paramerter but use policy to auto-set the header.
+""")
+  @doc("""
+Standard request id header.
+""")
+  @post
+  @route("/standard")
+  op standard(
+    @header("x-ms-request-id")
```

Comment:
`x-ms-client-request-id` ?

Reply:
I am not sure whether you want to put standard client-request-id together with non-standard client-request-id in same service/client.

As there is only one `RequestIdPolicy`, (I assume) it can only handle one case for client-request-id.

Reply:
I agree that one client with one request id policy. Thus, I use multi-client to bypass this problem. I don't know if there is better solution.

Reply:
Just ignore the standard one? As it already tested in traits.

Even multiple-client might have some implication for different codegen, as they are required to do per-client configure on this RequestIdPolicy. (knowing this, Java should be able to do it, but not sure about other codegen)